File: day6/puzzle2.py
def main():
    with open("puzzle_input.txt", "rt") as f:
        for line in f:
            if line.startswith("Time:"):
                time = int(line[5:].strip().replace(" ", ""))
            elif line.startswith("Distance:"):
                distance = int(line[9:].strip().replace(" ", ""))
            else:
                continue
    options = 0
    for hold_time in range(time):
        speed = hold_time
        d = (time - hold_time) * speed
        if d > distance:
            options += 1
    print(f"options = {options}")


if __name__ == "__main__":
    main()

File: day6/puzzle2.lua
local function main()
    local f = io.open("puzzle_input.txt", "r")
    if not f then
        os.exit(1)
    end
    local time = nil
    local distance = nil
    for line in f:lines() do
        if line:sub(1, 5) == "Time:" then
            local temp = ""
            for value in line:sub(6, -1):gmatch("%d+") do
                temp = temp .. value
            end
            time = tonumber(temp)
        elseif line:sub(1, 9) == "Distance:" then
            local temp = ""
            for value in line:sub(6, -1):gmatch("%d+") do
                temp = temp .. value
            end
            distance = tonumber(temp)
        end
    end
    f:close()
    local options = 0
    for hold_time = 0, time - 1 do
        local speed = hold_time
        local d = (time - hold_time) * speed
        if d > distance then
            options = options + 1
        end
    end
    print(string.format("options = %d", options))
end

main()
